src/Components/Molecules/sub-menu/sub-menu.tsx

```diff
@@ -0,0 +1,34 @@
+import "../sub-menu/sub-menu.scss";
+import { BackButtonTitle, MenuData } from "../../../data";
+import SubMenuBack from "../../Attoms/subMenu-back-button/submenu-back";
+
+const SubMenu = (props: MenuData & BackButtonTitle) => {
```

Comment:
avoid defining component props with intersection type as the merged types can have conflict in properties which will resolve to `never` type
```
interface SubMenuProps {
menu: MenuItem[];
closeSubMenu: () => void;
title: string | undefined;
buttonBackPath: string;
buttonBackAltTag: string;
}
```

or for more robust approach
```
interface SubMenuProps {
menuData: MenuData;
ackButton: BackButtonTitle
}
```